go/vt/vtgate/executor_select_test.go

```diff
@@ -2329,7 +2329,7 @@ func TestSelectScatterLimit(t *testing.T) {
 	require.NoError(t, err)
 
 	wantQueries := []*querypb.BoundQuery{{
-		Sql:           "select col1, col2, weight_string(col2) from `user` order by `user`.col2 desc limit :__upper_limit",
+		Sql:           "select col1, col2, weight_string(col2) from `user` order by `user`.col2 desc limit 3",
 		BindVariables: map[string]*querypb.BindVariable{"__upper_limit": sqltypes.Int64BindVariable(3)},
```

Comment:
Do we need the bind variable since we are removing the use of `:__upper_limit` above? cc @harshit-gangal

Reply:
We need to track whether the bind variable is part of the query or not.

Reply:
@systay any plan to do this?